### PR TITLE
HPCC-19565 Remote disk read support in Thor

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - not for merging yet, and based on your HPCC-19229 (+includes a fix in yours I think needed).

but could be reviewed.

**Comment by jakesmith** [comment]
@ghalliday - ready for review

**Comment by AttilaVamos** [comment]
@jakesmith Sorry, that was my fault, rescheduled.

**Comment by ghalliday** [comment]
@AttilaVamos (and others!) if you mention someone then it will be assigned to that person in the pull summary.  That means whoever was asked to review does not have it appear in their list  any more.  So if you mention jake, please mention @ghalliday afterwards so it stays in my list.

**Comment by AttilaVamos** [comment]
@ghalliday all right. Sorry

File: thorlcr/activities/diskread/thdiskreadslave.cpp [hunk 1]
```diff
@@ -250,46 +256,95 @@ void CDiskRecordPartHandler::open()
...
+
+            StringBuffer path;
+            if (!rfn.isLocal() || testForceRemote(rfn.getLocalPath(path)))
+            {
+                // Open a stream from remote file, having passed actual, expected, projected, and filters to it
+                SocketEndpoint ep(rfn.queryEndpoint());
+                setDafsEndpointPort(ep);
+
+                RowFilter actualFilter;
+                Owned<const IKeyTranslator> keyedTranslator = createKeyTranslator(actualFormat->queryRecordAccessor(true), expectedFormat->queryRecordAccessor(true));
```
Comment (ghalliday) [inline thorlcr/activities/diskread/thdiskreadslave.cpp]: could avoid creating the translator if activity.fieldFilters is empty.

Comment (ghalliday) [reply thorlcr/activities/diskread/thdiskreadslave.cpp]: ( also if actual = projected, but pointers are unlikely to be equal at the moment...)

Comment (jakesmith) [reply thorlcr/activities/diskread/thdiskreadslave.cpp]: ok, have pushed a commit a avoid if not necessary.

**Comment by jakesmith** [comment]
@ghalliday - pushed commit to re. keyed translator

**Pull request merged**